# `no-curly-quotes`

<kbd>Recommended âœ…</kbd> <kbd>Fixable ğŸ› ï¸</kbd>

Disallow curly quotes(`â€œ`, `â€`, `â€˜` or `â€™`) in text

## Rule Details

The purpose of this rule is to allow the use of ASCII characters in the editor and optionally convert them to curly symbols during rendering, using a parser that supports this feature, such as the Typographer extension of [Goldmark](https://github.com/yuin/goldmark) or [SmartyPants](https://daringfireball.net/projects/smartypants/).

Additionally, curly quotes (`â€œ`(`\u201C`), `â€`(`\u201D`), `â€˜`(`\u2018`) or `â€™`(`\u2019`)), which are often introduced by word processors like Word, Google Docs, and Pages, can cause unwanted issues in code and markup. This rule helps keep your code clean and consistent by preventing unintended curly quotes.

This rule only checks for curly quotes and applies only to [`text`](https://github.com/syntax-tree/mdast?tab=readme-ov-file#text) node.
### âŒ Incorrect

Examples of **incorrect** code for this rule:

```md
â€œfoo barâ€
â€˜foo barâ€™
```

### âœ… Correct

Examples of **correct** code for this rule:

```md
"foo bar"
'foo bar'
```

## Fix

This rule fixes the curly quotes by replacing them with straight quotes.

## Prior Art

- [textlint-rule-no-curly-quotes](https://github.com/aborazmeh/textlint-rule-no-curly-quotes)
